### ISSUE TITLE
Disabling MessageDrivenSubscriptions on transports with no native pubsub causes subscription requests to silently fail

### ISSUE THREAD

**Issue by andreasohlund** [Particular/NServiceBus#5120]
As described in https://github.com/Particular/NServiceBus/issues/5119 calling `.DisableFeature<MessageDrivenSubscriptions >()` for a transport that needs message driven pubsub causes subscription messages to not be sent out without any exception and error in the logs.

### Potential fixes

* Detect and throw at startup
* Make the  `MessageDrivenSubscriptions ` internal so it can't be disabled (breaking change)

**Comment by SimonCropp** [comment]
isnt this a message loss scenario? and hence should be given some priority

**Comment by andreasohlund** [comment]
I and @timbussmann looked at this and while we can do a better job in highlighting the dangers of this, `.DisableFeature<MessageDrivenSubscriptions >()` does exactly what it says => it turns the subscriptions off.  So we don't see this as a message loss bug.

We do think that we can do more here: (potentially in the upcoming 7.2)

* Add a default terminator that throws if auto-subscribe is on but the feature is off
* Provide an explicit "Subscriber-only" mode which would not need storage to be defined but still sends out subscription messages.

What do you think?

**Comment by SimonCropp** [comment]
> does exactly what it says => it turns the subscriptions off. So we don't see this as a message loss bug.

Seems a strange stance to take. it would seem if someone stuffs up the config (https://github.com/Particular/NServiceBus/issues/5119) defaulting to a behavior that hides that scenario would be the "pit of failure"? 

For me this is the equivalent of "someone has disabled timeouts, so when they explicitly request a timeout, it is ignored". yes it is technically correct, but i cant understand who would desire that

**Comment by andreasohlund** [comment]
I agree that we should fix this but it won't be treated as an urgent issue including backports etc. We have a 7.2 coming up I'll try to make sure this gets included, stay tuned

**Comment by SimonCropp** [comment]
@andreasohlund i wasnt expecting backports. but i was expecting it to be fixed in under 12 months

**Comment by timbussmann** [comment]
fyi, the pipeline already throws an exception today when trying to subscribe with `MessageDrivenSubscriptions`. However, the auto-subscribe feature does swallow exceptions when subscribing (but logs them!). As part of a minor version we decided to not change the behavior of auto-subscribe as there are too many potential issues this can cause to existing users. Instead, we improved the situations by removing the need to disable the MessageDrivenSubscriptions feature. The main reason to disable this feature seems to be to avoid providing a subscription storage, therefore a new `DisablePublishing()` configuration API is available on the transport configuration. Disabling publishing removes the need to provide a subscription storage, subscribe/unsubscribe is  still possible.
